### PR TITLE
chore(clippy): apply idiomatic changes

### PR DESCRIPTION
**Pull request by robertohuertasm**
## What problem are you trying to solve?

There are some changes that could be done to improve readability in the code, avoid allocations or simplify the code.

## What is your solution?

By following some of the Clippy recommendations I decided to apply some refactors to the codebase. I'll try to explain the rationale of each change in the code itself by leaving some messages as a mode of auto-review.

## What the reviewer should know

No problem if some change is deemed as subjective and you prefer to dismiss it.

### CODE REVIEW

**Comment by juli1** [comment]
Thanks for sending a pull request!

First, as for the previous pull request, is it possible to follow the template and explain the problem?

Second, this PR changes a lot of files with some changes being subjective (e.g. using `println!("Running {:?}", command);` or `println!("Running {command:?}");`). Our objective is to have pull request that focuses on a clear problem at hand and avoid too many small changes that are subjective and not a core issue.

In short, is it possible to explain the problem this PR solves (or the feature it brings) and focuses on this?

Thanks!

File: Cargo.toml [hunk 1]
```diff
@@ -4,6 +4,7 @@ members = [
     "bins",
     "cli",
     "kernel",
+    "server"
```
Comment (robertohuertasm) [inline Cargo.toml:7]: Explicitly adding `server` to the workspace. It's already included via path dependency but for clarity's sake I think it was worth it to introduce it here.

File: kernel/build.rs [hunk 2]
```diff
@@ -6,11 +6,12 @@ where
     F: FnMut(&mut Command) -> &mut Command,
 {
     let mut command = Command::new(name);
-    println!("Running {:?}", command);
+    println!("Running {command:?}");
```
Comment (robertohuertasm) [inline kernel/build.rs:9]: See [clippy rule](https://rust-lang.github.io/rust-clippy/master/index.html#/uninlined_format_args). Readability change. Applies to all the changes like this.

File: kernel/build.rs [hunk 3]
```diff
@@ -6,11 +6,12 @@ where
...
     let configured = configure(&mut command);
-    if configured.status().is_err() {
-        panic!("failed to execute {:?}", configured);
-    }
+    assert!(
```
Comment (robertohuertasm) [inline kernel/build.rs:11]: Readability change. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/manual_assert). Applies to all if-panic in the codebase.

File: kernel/src/analysis/analyze.rs [hunk 4]
```diff
@@ -47,70 +47,72 @@ pub fn analyze(
...
-                let invalid_query_result = RuleResult {
-                    rule_name: rule.name.clone(),
-                    filename: filename.to_string(),
-                    violations: vec![],
-                    errors: vec![ERROR_INVALID_QUERY.to_string()],
-                    execution_error: None,
-                    execution_time_ms: 0,
-                    output: None,
-                };
+    get_tree(code, language).map_or_else(
```
Comment (robertohuertasm) [inline kernel/src/analysis/analyze.rs:50]: This one is possibly the most subjective one although it's also based on some clippy lints like [this](https://rust-lang.github.io/rust-clippy/master/index.html#/option_if_let_else) and [this one](https://rust-lang.github.io/rust-clippy/master/index.html#/option_if_let_else).

File: kernel/src/analysis/javascript.rs [hunk 5]
```diff
@@ -133,7 +133,7 @@ lazy_static! {
     }
         "#,
         );
-        let mut rt = JsRuntimeForSnapshot::new(Default::default(), Default::default());
+        let mut rt = JsRuntimeForSnapshot::new(RuntimeOptions::default(), Default::default());
```
Comment (robertohuertasm) [inline kernel/src/analysis/javascript.rs:136]: Readability change. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/default_trait_access).

File: kernel/src/analysis/javascript.rs [hunk 6]
```diff
@@ -418,10 +419,12 @@ res
                     rule.name, filename, e
                 );
             }
-            let error_message = match e.to_string().find("at rule_code") {
-                Some(pos) => e.to_string()[..pos].to_string(),
-                None => e.to_string(),
-            };
+
+            let err_str = e.to_string();
+            let error_message = err_str
```
Comment (robertohuertasm) [inline kernel/src/analysis/javascript.rs:424]: Readability and conciseness change. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/option_if_let_else).

File: kernel/src/analysis/tree_sitter.rs [hunk 7]
```diff
@@ -1,10 +1,10 @@
 use crate::model::analysis::{MatchNode, MatchNodeContext, TreeSitterNode};
 use crate::model::common::{Language, Position};
-use anyhow::{anyhow, Result};
+use anyhow::Result;
 use std::collections::HashMap;
 use tree_sitter::QueryCursor;
 
-fn get_tree_sitter_language(language: &Language) -> Option<tree_sitter::Language> {
+fn get_tree_sitter_language(language: &Language) -> tree_sitter::Language {
```
Comment (robertohuertasm) [inline kernel/src/analysis/tree_sitter.rs:7]: Here, there's no need to return the `Option` at all, given that the code is always going to return a `Some` variant. This is introducing unnecessary complexity.

Comment (robertohuertasm) [reply kernel/src/analysis/tree_sitter.rs:7]: More context [here](https://rust-lang.github.io/rust-clippy/master/index.html#/unnecessary_wraps)

File: kernel/src/analysis/tree_sitter.rs [hunk 8]
```diff
@@ -129,12 +124,14 @@ pub fn map_node(node: tree_sitter::Node) -> Option<TreeSitterNode> {
                 line: u32::try_from(cursor.node().range().end_point.row + 1).unwrap(),
                 col: u32::try_from(cursor.node().range().end_point.column + 1).unwrap(),
             },
-            field_name: cursor.field_name().map(|v| v.to_string()),
+            field_name: cursor.field_name().map(ToString::to_string),
```
Comment (robertohuertasm) [inline kernel/src/analysis/tree_sitter.rs:127]: Typical [refactor](https://rust-lang.github.io/rust-clippy/master/index.html#/redundant_closure_for_method_calls) to avoid unnecessary closures.

File: kernel/src/analysis/tree_sitter.rs [hunk 9]
```diff
@@ -129,12 +124,14 @@ pub fn map_node(node: tree_sitter::Node) -> Option<TreeSitterNode> {
...
             children,
         };
 
         Some(ts_node)
     }
+
+    let mut ts_cursor = node.walk();
```
Comment (robertohuertasm) [inline kernel/src/analysis/tree_sitter.rs:134]: Move this line after the internal function for readability's sake. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/items_after_statements).

File: kernel/src/analysis/tree_sitter.rs [hunk 10]
```diff
@@ -84,7 +81,7 @@ pub fn get_query_nodes(
                 filename: filename.to_string(),
                 variables: variables.clone(),
             },
-        })
+        });
```
Comment (robertohuertasm) [inline kernel/src/analysis/tree_sitter.rs:84]: Added the semicolon given that we're not returning. See [this](https://rust-lang.github.io/rust-clippy/master/index.html#/semicolon_if_nothing_returned).

File: kernel/src/model/rule.rs [hunk 11]
```diff
@@ -162,7 +162,7 @@ impl Rule {
         }
         let description = self
             .decode_description()
-            .unwrap_or(Some("invalid description".to_string()));
+            .unwrap_or_else(|_| Some("invalid description".to_string()));
```
Comment (robertohuertasm) [inline kernel/src/model/rule.rs:165]: Refactor to avoid allocation in case the unwrap goes well. See this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/or_fun_call).

File: kernel/src/model/common.rs [hunk 12]
```diff
@@ -2,7 +2,7 @@ use derive_builder::Builder;
 use serde::{Deserialize, Serialize};
 use std::fmt;
 
-#[derive(Deserialize, Debug, Serialize, PartialEq)]
+#[derive(Deserialize, Debug, Serialize, Eq, PartialEq)]
```
Comment (robertohuertasm) [inline kernel/src/model/common.rs:5]: Added `Eq` trait as per this [clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/derive_partial_eq_without_eq).

File: kernel/src/model/common.rs [hunk 13]
```diff
@@ -11,10 +11,10 @@ pub enum OutputFormat {
 impl fmt::Display for OutputFormat {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let s = match self {
-            OutputFormat::Json => "JSON",
-            OutputFormat::Sarif => "SARIF",
+            Self::Json => "JSON",
```
Comment (robertohuertasm) [inline kernel/src/model/common.rs:14]: It's usually a good practice to use `Self` in these scenarios. See  [this](https://rust-lang.github.io/rust-clippy/master/index.html#/use_self).

**Comment by robertohuertasm** [comment]
@juli1 I've commented all the changes so you can judge on each of them. There's a rationale attached to each of them most of the time based on readability and common practices.

File: kernel/src/analysis/analyze.rs (hunk 4 above)
Comment (juli1) [reply kernel/src/analysis/analyze.rs:50]: I agree with you, this one does not really make the code any better. I would tend to not include this change as its quite hard to read. An alternative would be to change the mapping code and divide it into functions to make it more digestible.

**Review by juli1 — APPROVED** [review]
Thanks for taking the time to add explanation to the changes! This looks good to me. One caveat on the change in `analyze` as it makes the code slightly harder to read but we can ship this and refactor later.
